### PR TITLE
total improvement on /z aliases

### PR DESCRIPTION
**Pull request by Mantou1233**
fixed that you cannot use /z for autocorrect and error when deploying subcommands with no related z command name. (it will ignore it.)

### CODE REVIEW

File: src/classes/AutocompleteManager.ts [hunk 1]
```diff
@@ -82,7 +83,19 @@ export class AutocompleteManager {
     if (this.client.devMode && interaction.guild.id !== constant.mainGuild.id) return;
 
     let commandName = interaction.commandName;
-    if (interaction.options.getSubcommand(false)) commandName += '_' + interaction.options.getSubcommand(false)
+    if (commandName == "z") {
+      // 因爲z指令都是subcommands，所以getSubCommand是required
```
Comment (Zollo757347) [inline src/classes/AutocompleteManager.ts]:
```suggestion
    if (commandName === "z") {
      // 因爲 z 指令都是 subcommands，所以 getSubCommand 是 required
```

File: src/classes/AutocompleteManager.ts [hunk 2]
```diff
@@ -82,7 +83,19 @@ export class AutocompleteManager {
...
+
+      let zAlias = Translator.getFullCommandName(subcommand);
+
+      if (!zAlias) return interaction.respond([]);
+      
+      commandName = zAlias.join("_");
+    }
+    else {
+      if (interaction.options.getSubcommand(false)) commandName += '_' + interaction.options.getSubcommand(false)
+    }
```
Comment (Zollo757347) [inline src/classes/AutocompleteManager.ts:98]:
```suggestion
    else if (interaction.options.getSubcommand(false)) {
      commandName += '_' + interaction.options.getSubcommand(false);
    }
```

File: src/deploy.ts [hunk 3]
```diff
@@ -120,7 +120,9 @@ function loadCommands(dirPath: string): {
       .setDescription('執行 z 指令')
       .setDMPermission(false);
     zGlobalCommands.forEach(([parent, command]) => {
-      command.setName(Translator.getZShortcut([parent, command.name])!);
+      const shortcut = Translator.getZShortcut([parent, command.name])
+      if(!shortcut) return;
+      command.setName(shortcut);
```
Comment (Zollo757347) [inline src/deploy.ts:125]:
```suggestion
      const shortcut = Translator.getZShortcut([parent, command.name]);
      if (!shortcut) return;
      command.setName(shortcut);
```

**Review by Zollo757347 — APPROVED** [review]

File: src/classes/AutocompleteManager.ts [hunk 4]
```diff
@@ -82,7 +83,19 @@ export class AutocompleteManager {
     if (this.client.devMode && interaction.guild.id !== constant.mainGuild.id) return;
 
     let commandName = interaction.commandName;
-    if (interaction.options.getSubcommand(false)) commandName += '_' + interaction.options.getSubcommand(false)
+    if (commandName === "z") {
+      // 因爲z指令都是subcommands，所以getSubCommand是required
```
Comment (kyometori) [inline src/classes/AutocompleteManager.ts:87]:
```suggestion
      // 因為 z 指令都是群組指令，所以 getSubcommand 是 required
```

**Review by kyometori — CHANGES_REQUESTED** [review]
我想註解風格也要規範一下

**Comment by kyometori** [comment]
笑死 @Zollo757347 好像有 request 過，但被忽略了

**Review by kyometori — APPROVED** [review]

**Pull request merged**